.github/workflows/docs.yml

```diff
@@ -0,0 +1,24 @@
+---
+name: s2n-tls Documentation
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  generate-doxygen:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Create Documentation
+        run: |
+            .github/s2n_doxygen.sh
+      - name: Upload Documentation
+        uses: actions/upload-artifact@v3
```

Comment:
There is a pages specific up-loader that might be less work, unless there is some security issue? https://github.com/peaceiris/actions-gh-pages

Reply:
Oh this is super cool. I think we could use this to push the docs to the `gh-pages` on merge. 

`gh-pages` could contain the bleeding edge documentation and release tags could contain lock step version documentation